spec/models/user_spec.rb

```diff
@@ -271,4 +271,75 @@
       expect(duplicates).to eq(3)
     end
   end
+
+  describe "password_resets_limit_reached?" do
+    context "user with no password reset requests" do
+      let(:user) { build(:user, resets_requested: 0) }
+
+      it "has not reached the requests limit" do
+        expect(user.password_resets_limit_reached?).to be_falsy
+      end
+    end
+
+    context "user at the limit of password resets but after cooldown period" do
+      let(:user) do
+        build(:user,
+              resets_requested: ArchiveConfig.PASSWORD_RESET_LIMIT,
+              reset_password_sent_at: ArchiveConfig.PASSWORD_RESET_COOLDOWN_HOURS.hours.ago)
+      end
+
+      it "has not reached the requests limit" do
+        expect(user.password_resets_limit_reached?).to be_falsy
+      end
+    end
+
+    context "user at the limit of password resets and within the cooldown period" do
+      let(:user) do
+        build(:user,
+              resets_requested: ArchiveConfig.PASSWORD_RESET_LIMIT,
+              reset_password_sent_at: Time.current)
+      end
+
+      it "has reached the requests limit" do
+        expect(user.password_resets_limit_reached?).to be_truthy
+      end
+    end
```

Comment:
We can probably better leverage `context` here:
```suggestion
    context "with the maximum number of resets requested" do
      context "when the cooldown period has passed" do
        let(:user) do
          build(:user,
                resets_requested: ArchiveConfig.PASSWORD_RESET_LIMIT,
                reset_password_sent_at: ArchiveConfig.PASSWORD_RESET_COOLDOWN_HOURS.hours.ago)
        end

        it "has not reached the requests limit" do
          expect(user.password_resets_limit_reached?).to be_falsy
        end
      end

      context "when the cooldown period has not passed" do
        let(:user) do
          build(:user,
                resets_requested: ArchiveConfig.PASSWORD_RESET_LIMIT,
                reset_password_sent_at: Time.current)
        end

        it "has reached the requests limit" do
          expect(user.password_resets_limit_reached?).to be_truthy
        end
      end
    end
```
You might even be able to just define `user` within the initial `context` block and use a `before` to update `reset_password_sent_at` to make it even shorter, but I don't have my dev environment fired up to run the test, so I didn't want to get too involved.